xray/segment.go

```diff
@@ -54,6 +54,7 @@ func noOpSegmentID() string {
 }
 
 // BeginFacadeSegment creates a Segment for a given name and context.
+// Caution: This is an internal API only to be used in Lambda context within the SDK. Consider using BeginSegment instead.
```

Comment:
Nit: Maybe can use `Note` instead of `Caution` ?

Reply:
+1, having `NOTE` specifically usually gets picked up by text editors, i.e. :

![image](https://user-images.githubusercontent.com/23139455/146614144-a3147441-5cb6-4995-a677-77dcbee30e8c.png)